CMakeLists.txt

```diff
@@ -31,118 +31,198 @@ endif(NOT WIN32)
 
 add_subdirectory(thirdparty)
 
-set(sources
-    movieplayback/LFD_Film.cpp
-    movieplayback/MovieManager.cpp
-    ui/UI_System.cpp
-    ui/XL_Console.cpp
-    ui/Console.cpp
-    scriptsystem/ScriptSystem.cpp
-    scriptsystem/scriptarray.cpp
-    scriptsystem/scriptstdstring.cpp
-    procedural/ProceduralFunc.cpp
-    procedural/Noise.cpp
-    os/Input.cpp
-    EngineSettings.cpp
-    plugin_framework/DynamicLibrary.cpp
-    plugin_framework/PluginManager.cpp
-    Engine.cpp
-    math/Matrix.cpp
-    math/Vector2.cpp
-    math/Plane.cpp
-    math/Vector4.cpp
-    math/Vector3.cpp
-    math/FixedPoint.cpp
-    math/Math.cpp
-    fileformats/TextureConv_IMG.cpp
-    fileformats/LFD_Reader.cpp
-    fileformats/ArchiveManager.cpp
-    fileformats/LFD_Anim.cpp
-    fileformats/CellManager.cpp
-    fileformats/MeshLoader_Daggerfall.cpp
-    fileformats/CellLoader_OutlawsMap.cpp
-    fileformats/Location_Daggerfall.cpp
-    fileformats/TextureConv_PCX.cpp
-    fileformats/GOB_Reader.cpp
-    fileformats/BSA_Reader.cpp
-    fileformats/RFF_Reader.cpp
-    fileformats/TextureLoader.cpp
-    fileformats/SkyLoader_Daggerfall.cpp
-    fileformats/ART_Reader.cpp
-    fileformats/CellLoader_BloodMap.cpp
-    fileformats/CellLoader_Daggerfall.cpp
-    fileformats/DFFaceTex.cpp
-    fileformats/TextureConv_ART.cpp
-    fileformats/Parser.cpp
-    world/LevelFunc_Default.cpp
-    world/Terrain.cpp
-    world/Sprite_ZAxis.cpp
-    world/Logic.cpp
-    world/ObjectManager.cpp
-    world/World.cpp
-    world/WorldCell.cpp
-    world/MeshCollision.cpp
-    world/Sector_2_5D.cpp
-    world/OrientedSprite.cpp
-    world/LogicManager.cpp
-    world/Object.cpp
-    world/Sector.cpp
-    world/LevelFunc.cpp
-    world/LevelFuncMgr.cpp
-    world/Sector_GeoBlock.cpp
-    render/CommandBuffer.cpp
-    render/Font.cpp
-    render/RenderQue.cpp
-    render/FontManager.cpp
-    render/MeshCache.cpp
-    render/Driver3D_Soft.cpp
-    render/ImageLoader.cpp
-    render/Camera.cpp
-    render/TextureCache.cpp
-    render/Driver3D_OGL.cpp
-    render/IDriver3D.cpp
-    render/VertexBuffer.cpp
-    render/triangleRasterizer/TriangleRasterizer.cpp
-    render/triangleRasterizer/DrawScanline.cpp
-    render/Mesh.cpp
-    render/IndexBuffer.cpp
-    networking/NetworkMgr.cpp
-    memory/ScratchPad.cpp
-)
+set(src_root ${XLEngine_SOURCE_DIR})
 
+# Organize XL Engine sources into alphabetized groups.
+# Can't use GLOB_RECURSE everywhere because some files are platform-specific.
+file(GLOB_RECURSE fileformats_sources
+    ${src_root}/fileformats/*.h*
+    ${src_root}/fileformats/*.cpp)
```

Comment:
I'd prefer not using glob because it doesn't create a dependency on new or removed files. So if I add or delete a source in `fileformats/` for instance, it won't automatically re-trigger cmake to include or exclude that source in the build, and cause build errors until cmake is manually invoked. With the files explicitly listed, you have to edit `CMakeLists.txt` for the change, which will implicitly cause cmake to rerun on the next build attempt.

Reply:
I've always just been manually invoking the CMake configure/generate, so I guess that's not the best way to do things after all ;). I can change them to not use `GLOB_RECURSE`.